api/redhatopenshift/HcpCluster.Management/hcpCluster-models.tsp

```diff
@@ -346,6 +350,77 @@ union OutboundType {
   loadBalancer: "loadBalancer",
 }
 
+/** The configuration that the operators of the cluster have to authenticate to Azure. */
+model OperatorsAuthenticationProfile {
+  /** Represents the information related to Azure User-Assigned managed identities needed
+   * to perform Operators authentication based on Azure User-Assigned Managed Identities */
+  managedIdentities: ManagedIdentitiesProfile;
+}
+
+/** Represents the information related to Azure User-Assigned managed identities needed
+ * to perform Operators authentication based on Azure User-Assigned Managed Identities */
+model ManagedIdentitiesProfile {
+  /** The set of Azure User-Assigned Managed Identities leveraged for the Control Plane
+   * operators of the cluster. The set of required managed identities is dependent on the
+   * Cluster's OpenShift version. */
+  @OpenAPI.extension("x-ms-identifiers", ["operatorName"])
+  controlPlaneOperators: ControlPlaneOperatorsElement[];
+
+  /** The set of Azure User-Assigned Managed Identities leveraged for the Data Plane
+   * operators of the cluster. The set of required managed identities is dependent on the
+   * Cluster's OpenShift version. */
+  @OpenAPI.extension("x-ms-identifiers", ["operatorName"])
+  dataPlaneOperators: DataPlaneOperatorsElement[];
+
+  /** The Managed Identities Data Plane Identity URL associated with the cluster. It is the URL
+   * that will be used to communicate with the Managed Identities Resource Provider (MI RP). */
+  @visibility("create", "read")
+  dataPlaneIdentityUrl: url;
+
+  /** Represents the information associated to an Azure User-Assigned Managed Identity whose
+   * purpose is to perform service level actions. */
+  @visibility("create", "read")
+  serviceManagedIdentity: UserAssignedManagedIdentity;
+}
+
+/** Pairs a control plane operator name with its user assigned identity */
+model ControlPlaneOperatorsElement {
+  /** The control plane operator name */
+  operatorName: string;
+
+  ...UserAssignedManagedIdentity;
```

Comment:
the client-id and principal-id will be duplicated in both the `identity{}` field and this field if we choose to specify it here. 

```
properties: {
    "managedIdentityProfile": {
        "controlPlaneOperators": [
             {
             "operatorName": "csi-disk-driver", 
             "resourceId": "/subscriptions/.../Microsoft.Identity/managedIdentity/csi-disk-driver"
             "principalId": "csi-disk-driver",
             "clientId": "csi-disk-driver"
             }, 
             {...}
         ]
    }   
},
identity: {
    "/subscriptions/.../Microsoft.Identity/managedIdentity/csi-disk-driver": {
        "principalId": "csi-disk-driver",
        "clientId": "csi-disk-driver"
    }, 
    "/subscriptions/.../csi-file-driver": {
        "principalId": "csi-file-driver",
        "clientId": "csi-file-driver"
    }
}

Reply:
Ah, I had no idea what the top-level `identity` section would contain.  In that case we can have both data and control plane operators be a map of `operatorName -> resourceId` where `resourceId` is a lookup key for the `identity` section.

Reply:
Also, maybe the `serviceManagedIdentity` field just needs to be a resource ID lookup key as well?

Reply:
Just to call out here, the RP does not currently parse the top-level `identity` section at all so we'll need to get that implemented immediately in addition to the plumbing needed for this new API.